include/nbl/asset/CCPUMeshPackerV1.h

```diff
@@ -496,6 +506,122 @@ MeshPackerBase::PackedMeshBufferData CCPUMeshPacker<MDIStructType>::commit(const
 	return { ramb.mdiAllocationOffset, MDIStructsAddedCnt };
 }
 
+template <typename MDIStructType>
+static bool CCPUMeshPackerV1<MDIStructType>::cmpVtxInputParams(const SVertexInputParams& lhs, const SVertexInputParams& rhs)
+{
+	if (lhs.enabledAttribFlags != rhs.enabledAttribFlags)
+		return false;
+
+	for (uint16_t attrBit = 0x0001, location = 0; location < SVertexInputParams::MAX_ATTR_BUF_BINDING_COUNT; attrBit <<= 1, location++)
+	{
+		if (!(attrBit & lhs.enabledAttribFlags))
+			continue;
+
+		if (lhs.attributes[location].format != rhs.attributes[location].format ||
+			lhs.bindings[lhs.attributes[location].binding].inputRate != rhs.bindings[rhs.attributes[location].binding].inputRate)
+			return false;
+	}
+
+	return true;
+}
+
+template <typename MDIStructType>
+template <typename Iterator>
+static uint32_t CCPUMeshPackerV1<MDIStructType>::getPackerCreationParamsFromMeshBufferRange(const Iterator begin, const Iterator end, Iterator sortedMeshBuffersOut,
+	MeshPackerConfigParams<Iterator>* packerParamsOut)
+{
+	assert(begin <= end);
+	if (begin == end)
+		return 0;
+
+	uint32_t packersNeeded = 1u;
+
+	IMeshPackerBase::MeshPackerConfigParams<Iterator> firstInpuParams
+	{
+		(*begin)->getPipeline()->getVertexInputParams(),
+		SRange<void, Iterator>(sortedMeshBuffersOut, sortedMeshBuffersOut)
+	};
+	memcpy(packerParamsOut, &firstInpuParams, sizeof(SVertexInputParams));
+
+	//fill array
+	auto test1 = std::distance(begin, end);
+	auto* packerParamsOutEnd = packerParamsOut + 1u;
+	for (Iterator it = begin + 1; it != end; it++)
+	{
+		auto& currMeshVtxInputParams = (*it)->getPipeline()->getVertexInputParams();
+
+		bool alreadyInserted = false;
+		for (auto* packerParamsIt = packerParamsOut; packerParamsIt != packerParamsOutEnd; packerParamsIt++)
```

Comment:
@Crisspl double nested `for` check if it doesn't turn the problem into O(n^2)

seems like this would be better solved with returning a `core::unordered_multimap<MeshPackerConfigParams,MeshIterator>`